newIDE/app/src/Utils/i18n/GDI18nProvider.js

```diff
@@ -39,41 +39,47 @@ export default class GDI18nProvider extends React.Component<Props, State> {
     }
   }
 
-  _loadCatalog = (language: string): Promise<Catalogs> => {
+  _loadCatalog = async (language: string): Promise<Catalogs> => {
     if (this.state.catalogs[language]) {
-      return Promise.resolve(this.state.catalogs);
+      return this.state.catalogs;
     }
 
-    return import(/* webpackMode: "lazy", webpackChunkName: "locales-[request]" */
-    `../../locales/${language.replace('-', '_')}/messages`).then(
-      catalog => {
-        return { ...this.state.catalogs, [language]: catalog };
-      },
-      (error: Error) => {
-        console.error('Error while loading language=' + language, error);
-        return this.state.catalogs;
-      }
-    );
+    try {
+      const editorCatalog = await import(/* webpackMode: "lazy", webpackChunkName: "locales-[request]" */
+      `../../locales/${language.replace('-', '_')}/messages`);
+      const extensionCatalog = await import(/* webpackMode: "lazy", webpackChunkName: "locales-[request]" */
+      `../../locales/${language.replace('-', '_')}/extension-messages`);
+      const catalog = {
+        languageData: editorCatalog.languageData,
+        messages: {
+          ...editorCatalog.messages,
+          ...extensionCatalog.messages,
```

Comment:
Maybe the other way around so that editor catalog is overriding the other?

Reply:
indeed